functions/routes/discount-rules.js

```diff
@@ -0,0 +1,47 @@
+const getAppData = require('../lib/store-api/get-app-data')
+const { validateDateRange } = require('../lib/helpers')
+
+const isValidRule = (rule, domain) => {
+  const isInvalidRule = Boolean(
+    !validateDateRange(rule) || (rule.domain && rule.domain !== domain) ||
+    (!rule.product_ids?.length && !rule.category_ids?.length && !rule.excluded_product_ids?.length) ||
+    ((rule.utm_campaign || (rule.customer_ids && rule.customer_ids.length) || rule.discount_coupon))
+  )
+  return !isInvalidRule
+}
+
+exports.get = async ({ appSdk }, req, res) => {
+  const storeId = Number(req.query?.store_id || 0)
+  const origin = req.headers.origin
+  let domain = origin
+  if (!origin || origin?.includes('localhost') || origin?.includes('127.0.0.1')) {
+    domain = req.query?.domain
+  }
+
+  if (!domain) {
+    return res.status(400)
+      .send({ message: 'Parameter "domain" not found' })
+  }
+
+  domain = domain.replace(/https?:\/\/|:\d+/g, '')
```

Comment:
```suggestion
  }
  domain = domain.replace(/^(https?:\/\/)?([^:/]+).*/, '$2')
```

Remover possível path ou trailing slash